### PR TITLE
Update dynamic_programming/coin_change.py

### PR DESCRIPTION
**Pull request by rudrakshkarpe**
### Describe your change:
- Adding one of the famous problems of dynamic programming series i.e. coin change-making problem

* [X] Add an algorithm?
* [   ] Fix a bug or typo in an existing algorithm?
* [   ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanations.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by rudrakshkarpe** [comment]
I have done the required changes according to the guidelines, please let me know if anything missing. It's great please to contribute to The Alogrithms/Python

File: dynamic_programming/coin_change.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+# (7 + 7 + 3 + 1) or (5 + 5 + 5 + 3) or (7 + 5 + 5 + 1)
+
+
+import sys
+
+
+# Function to find the minimum number of coins required
+# to get a total of `target` from set `S`
+def Find_minimum_coins(S, target) -> int:
```
Comment (CaedenPH) [inline dynamic_programming/coin_change.py:15]:
```suggestion
def find_minimum_coins(S, target) -> int:
```

File: dynamic_programming/coin_change.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
```
Comment (CaedenPH) [inline dynamic_programming/coin_change.py:16]:
```suggestion
```

File: dynamic_programming/coin_change.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        # coin resulted in a solution
+        if result != sys.maxsize:
+            coins = min(coins, result + 1)
+
+    # return the minimum number of coins needed
+    return coins
+
+
+if __name__ == '__main__':
+
```
Comment (CaedenPH) [inline dynamic_programming/coin_change.py:49]:
```suggestion
```

File: dynamic_programming/coin_change.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    # return infinity if total becomes negative
+    if target < 0:
+        return sys.maxsize
+
+    # initialize the minimum number of coins needed to infinity
+    coins = sys.maxsize
+
+    # do for each coin
+    for c in S:
```
Comment (CaedenPH) [inline dynamic_programming/coin_change.py:34]: Include more descriptive variable names

File: dynamic_programming/coin_change.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# If the desired change is 15, the minimum number of coins required is 3
+
+# (7 + 7 + 1) or (5 + 5 + 5) or (3 + 5 + 7)
+
+# If the desired change is 18, the minimum number of coins required is 4
+
+# (7 + 7 + 3 + 1) or (5 + 5 + 5 + 3) or (7 + 5 + 5 + 1)
```
Comment (CaedenPH) [inline dynamic_programming/coin_change.py:7]:
```suggestion
"""
If the desired change is 15, the minimum number of coins required is 3

(7 + 7 + 1) or (5 + 5 + 5) or (3 + 5 + 7)

If the desired change is 18, the minimum number of coins required is 4

(7 + 7 + 3 + 1) or (5 + 5 + 5 + 3) or (7 + 5 + 5 + 1)
"""
```

File: dynamic_programming/coin_change.py (hunk 4 above)
Comment (rudrakshkarpe) [reply dynamic_programming/coin_change.py:34]: Alright I'll do it asap

File: dynamic_programming/coin_change.py (hunk 1 above)
Comment (rudrakshkarpe) [reply dynamic_programming/coin_change.py:15]: Done!

File: dynamic_programming/coin_change.py (hunk 3 above)
Comment (rudrakshkarpe) [reply dynamic_programming/coin_change.py:49]: Removed extra space

File: dynamic_programming/coin_change.py (hunk 5 above)
Comment (rudrakshkarpe) [reply dynamic_programming/coin_change.py:7]: Docstrings added

**Comment by cclauss** [comment]
In preparation for that denial-of-service attack that is [Hacktoberfest](https://hacktoberfest.com), I am closing all pull requests that have failing tests.

**Pull request closed without merging**